README.md

```diff
@@ -2098,14 +2099,7 @@ set :protection, :except => [:path_traversal, :session_hijacking]
 ```
 
 By default, Sinatra will only set up session based protection if `:sessions`
-has been enabled. Sometimes you want to set up sessions on your own, though. In
-that case you can get it to set up session based protections by passing the
-`:session` option:
-
-```ruby
-use Rack::Session::Pool
-set :protection, :session => true
-```
+has been enabled.
```

Comment:
*nitpick* "have been enabled".

Also, this is still true if you use the middleware option.

I don't want to ignore that use-case as many applications probably are doing it that way, and likely read it from here.

We should consider the `README` a kind of spec, because many applications rely on the behavior they learned from it, so we can't simply remove it without notice.

IMO this would mean having different sections under the "Using Sessions" chapter. Likely, we would start out with `enable :session`, like we do here including the note about defaulting to a cookie store. Followed by a section on "Choosing your own session middleware", which outlines both use-cases you've identified above.

As well, the section below this "Available settings" might also need to be updated wrt the "sessions" option. WDYT?

tl;dr I'm happy to support both cases independently.